dask_kubernetes/operator/kubecluster/kubecluster.py

```diff
@@ -230,6 +230,23 @@ def __init__(
         if isinstance(self.worker_command, str):
             self.worker_command = self.worker_command.split(" ")
 
+        if self.n_workers is not None and not isinstance(self.n_workers, int):
+            raise TypeError(f"n_workers must be an integer, got {type(self.n_workers)}")
+
+        try:
+            # Validate `resources` param is a dictionary whose
+            # keys must either be 'limits' or 'requests'
+            assert isinstance(self.resources, dict)
+
+            for field in ("limits", "requests"):
+                if field in self.resources:
+                    assert isinstance(self.resources[field], dict)
+                elif field not in self.resources:
+                    raise ValueError(f"Unknown field '{field}' in resources")
```

Comment:
Can't seem to figure out why this line is causing all the failing tests errors, am I missing something? 

Here's an example output 
```
dask_kubernetes/operator/kubecluster/tests/test_kubecluster.py::test_kubecluster ERROR                                                                    [100%]

============================================================================ ERRORS =============================================================================
______________________________________________________________ ERROR at setup of test_kubecluster _______________________________________________________________

kopf_runner = <kopf._kits.runner.KopfRunner object at 0x7f9c366decd0>, docker_image = 'dask-kubernetes:dev'

    @pytest.fixture
    def cluster(kopf_runner, docker_image):
        with dask.config.set({"kubernetes.name": "foo-{uuid}"}):
            with kopf_runner:
>               with KubeCluster(image=docker_image, n_workers=1) as cluster:

dask_kubernetes/operator/kubecluster/tests/conftest.py:13: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <[AttributeError("'KubeCluster' object has no attribute '_cluster_info'") raised in repr()] KubeCluster object at 0x7f9c366bea00>, name = 'foo-{uuid}'
namespace = None, image = 'dask-kubernetes:dev', n_workers = 1, resources = None, env = None, worker_command = None
auth = [<dask_kubernetes.common.auth.InCluster object at 0x7f9c3baaa9d0>, <dask_kubernetes.common.auth.KubeConfig object at 0x7f9c3ba691f0>]

    def __init__(
        self,
        *,
        name=None,
        namespace=None,
        image=None,
        n_workers=None,
        resources=None,
        env=None,
        worker_command=None,
        auth=ClusterAuth.DEFAULT,
        port_forward_cluster_ip=None,
        create_mode=None,
        shutdown_on_close=None,
        resource_timeout=None,
        scheduler_service_type=None,
        custom_cluster_spec=None,
        scheduler_forward_port=None,
        **kwargs,
    ):
    
        name = dask.config.get("kubernetes.name", override_with=name)
        self.namespace = (
            dask.config.get("kubernetes.namespace", override_with=namespace)
            or get_current_namespace()
        )
        self.image = dask.config.get("kubernetes.image", override_with=image)
        self.n_workers = dask.config.get(
            "kubernetes.count.start", override_with=n_workers
        )
        if dask.config.get("kubernetes.count.max"):
            warnings.warn(
                "Setting a maximum number of workers is no longer supported. "
                "Please use Kubernetes Resource Quotas instead."
            )
        self.resources = dask.config.get(
            "kubernetes.resources", override_with=resources
        )
        self.env = dask.config.get("kubernetes.env", override_with=env)
        self.worker_command = dask.config.get(
            "kubernetes.worker-command", override_with=worker_command
        )
        self.auth = auth
        self.port_forward_cluster_ip = dask.config.get(
            "kubernetes.port-forward-cluster-ip", override_with=port_forward_cluster_ip
        )
        self.create_mode = dask.config.get(
            "kubernetes.create-mode", override_with=create_mode
        )
        self.shutdown_on_close = dask.config.get(
            "kubernetes.shutdown-on-close", override_with=shutdown_on_close
        )
        self._resource_timeout = dask.config.get(
            "kubernetes.resource-timeout", override_with=resource_timeout
        )
        self._custom_cluster_spec = dask.config.get(
            "kubernetes.custom-cluster-spec", override_with=custom_cluster_spec
        )
        self.scheduler_service_type = dask.config.get(
            "kubernetes.scheduler-service-type", override_with=scheduler_service_type
        )
        self.scheduler_forward_port = dask.config.get(
            "kubernetes.scheduler-forward-port", override_with=scheduler_forward_port
        )
    
        if self._custom_cluster_spec:
            if isinstance(self._custom_cluster_spec, str):
                with open(self._custom_cluster_spec) as f:
                    self._custom_cluster_spec = yaml.safe_load(f.read())
            name = self._custom_cluster_spec["metadata"]["name"]
    
        if isinstance(self.worker_command, str):
            self.worker_command = self.worker_command.split(" ")
    
        if self.n_workers is not None and not isinstance(self.n_workers, int):
            raise TypeError(f"n_workers must be an integer, got {type(self.n_workers)}")
    
        try:
            # Validate `resources` param is a dictionary whose
            # keys must either be 'limits' or 'requests'
            assert isinstance(self.resources, dict)
    
            for field in ("limits", "requests"):
                if field in self.resources:
                    assert isinstance(self.resources[field], dict)
                elif field not in self.resources:
>                   raise ValueError(f"Unknown field '{field}' in resources")
E                   ValueError: Unknown field 'limits' in resources

dask_kubernetes/operator/kubecluster/kubecluster.py:245: ValueError
```